src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -177,1327 +205,1182 @@ static std::vector<rct_widget> _window_editor_object_selection_widgets = {
-    events.get_scroll_size = &WindowEditorObjectSelectionScrollgetsize;
-    events.scroll_mousedown = &WindowEditorObjectSelectionScrollMousedown;
-    events.scroll_mouseover = &WindowEditorObjectSelectionScrollMouseover;
-    events.text_input = &WindowEditorObjectSelectionTextinput;
-    events.tooltip = &WindowEditorObjectSelectionTooltip;
-    events.invalidate = &WindowEditorObjectSelectionInvalidate;
-    events.paint = &WindowEditorObjectSelectionPaint;
-    events.scroll_paint = &WindowEditorObjectSelectionScrollpaint;
-});
-// clang-format on
-
-#pragma endregion
-
 static constexpr const int32_t window_editor_object_selection_animation_loops[] = {
-    20, 32, 10, 72, 24, 28, 16,
-};
+        20, 32, 10, 72, 24, 28, 16,
+    };
 static constexpr const int32_t window_editor_object_selection_animation_divisor[] = {
-    4, 8, 2, 4, 4, 4, 2,
-};
-
-static void WindowEditorObjectSetPage(rct_window* w, int32_t page);
-static void WindowEditorObjectSelectionSetPressedTab(rct_window* w);
-static int32_t GetObjectFromObjectSelection(ObjectType object_type, int32_t y);
-static void WindowEditorObjectSelectionManageTracks();
-static void EditorLoadSelectedObjects();
-static bool FilterSelected(uint8_t objectFlags);
-static bool FilterString(const ObjectRepositoryItem* item);
-static bool FilterSource(const ObjectRepositoryItem* item);
-static bool FilterChunks(const ObjectRepositoryItem* item);
-static void FilterUpdateCounts();
+        4, 8, 2, 4, 4, 4, 2,
+    };
 
-static std::string ObjectGetDescription(const Object* object);
-static ObjectType GetSelectedObjectType(rct_window* w);
+static rct_string_id GetRideTypeStringId(const ObjectRepositoryItem* item);
+static bool VisibleListSortRideType(const ObjectListItem& a, const ObjectListItem& b);
+static bool VisibleListSortRideName(const ObjectListItem& a, const ObjectListItem& b);
 
-enum
+class EditorObjectSelectionWindow final: public Window
 {
-    RIDE_SORT_TYPE,
-    RIDE_SORT_RIDE
-};
+using sortFunc_t = bool (*)(const ObjectListItem&, const ObjectListItem&);
+private:
+    std::vector<ObjectListItem> _listItems;
+    int32_t _listSortType = RIDE_SORT_TYPE;
+    bool _listSortDescending = false;
+    std::unique_ptr<Object> _loadedObject;
 
-enum
-{
-    DDIX_FILTER_RCT1,
-    DDIX_FILTER_AA,
-    DDIX_FILTER_LL,
-    DDIX_FILTER_RCT2,
-    DDIX_FILTER_WW,
-    DDIX_FILTER_TT,
-    DDIX_FILTER_OO,
-    DDIX_FILTER_CUSTOM,
-    DDIX_FILTER_SEPARATOR,
-    DDIX_FILTER_SELECTED,
-    DDIX_FILTER_NONSELECTED,
-};
 
-struct ObjectListItem
-{
-    const ObjectRepositoryItem* repositoryItem;
-    std::unique_ptr<rct_object_filters> filter;
-    uint8_t* flags;
-};
+public:
 
-static rct_string_id GetRideTypeStringId(const ObjectRepositoryItem* item);
+    /**
+     *
+     *  rct2: 0x006AA64E
+     */
+    void OnOpen() override
+    {
+        auto& targetWidgets = _window_editor_object_selection_widgets;
+        if (!_window_editor_object_selection_widgets_initialised)
```

Comment:
Can you revert this back into a separate private member function `InitWidgets `

Reply:
My bad, done https://github.com/OpenRCT2/OpenRCT2/pull/16027/commits/5ee572b27af9d2f01b935a74206500cbcad76b67